pkg/master/master.go

```diff
@@ -186,7 +191,12 @@ func NewEtcdHelper(client tools.EtcdGetSet, version string) (helper tools.EtcdHe
 	if err != nil {
 		return helper, err
 	}
-	return tools.NewEtcdHelper(client, versionInterfaces.Codec), nil
+	if prefix == "" {
+		prefix = DefaultEtcdPathPrefix
+	} else {
+		prefix = path.Join("/", prefix, DefaultEtcdPathPrefix)
```

Comment:
I insist that you remove this line, if someone passes in "/registry", this will use "/registry/registry". If someone passes a value for prefix IMO you should just use it directly.